src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs

```diff
@@ -208,10 +266,37 @@ private void AddNetStandardTargetingPackAssets(ITaskItem targetingPack, string t
             string targetingPackDllFolder,
             ITaskItem targetingPack, List<TaskItem> referenceItems, List<TaskItem> analyzerItems)
         {
-            XDocument frameworkListDoc = XDocument.Load(frameworkListPath);
-
             string profile = targetingPack.GetMetadata("Profile");
 
+            StringBuilder hashKey = new(nameof(ResolveTargetingPackAssets) + nameof(AddItemsFromFrameworkList));
+            hashKey.AppendLine();
+            hashKey.AppendLine(frameworkListPath);
+            hashKey.AppendLine(targetingPackRoot);
+            hashKey.AppendLine(targetingPackDllFolder);
+            hashKey.AppendLine(targetingPack.ItemSpec);
+            hashKey.AppendLine(profile);
+            hashKey.AppendLine(targetingPack.GetMetadata(MetadataKeys.NuGetPackageId));
+            hashKey.AppendLine(targetingPack.GetMetadata(MetadataKeys.NuGetPackageVersion));
+
+            string frameworkListKey = hashKey.ToString();
+
+            if (s_AllowCacheLookup &&
+                BuildEngine4.GetRegisteredTaskObject(
+                  frameworkListKey, 
+                  RegisteredTaskObjectLifetime.AppDomain /* TODO: should this be build? */) 
+                is FrameworkListCacheEntry cacheEntry)
+            {
+                // TODO: probe files to see if they've been modified? or just assume not
```

Comment:
The meta-question here is: is there any supported scenario where the contents of the targeting pack will change? Or can we cache it forever?

Reply:
Discussed this with @dsplaisted offline and we decided "process lifetime" is probably ok, as we don't expect targeting packs to change.